### DISCUSSION TITLE
question about cyberbots in mame_roms.xml (can't open samples and soundfont)

### DISCUSSION THREAD

**Discussion by EurekaSevven** [vgmtrans/vgmtrans#204] [Q&A]
This has been in my head for a while but I only made an account today to talk about it.
I have used VGMtrans for use with ripping instruments from other CPS2 games, in fact every one I've tried, but cybots has been the only exception, and I cannot figure out why.

This is exacerbated by the fact that all the offsets in the mame_roms.xml file for cybots seem to line up perfectly with the data inside my files.  The .01 file has "version 1.05A /CPS2--1993 / JUNE": 
The seq_table offset seems to be correct at 0x9005: 
![image](https://user-images.githubusercontent.com/82067599/113794504-b4027000-9718-11eb-88dc-f2dcc0609d31.png)
Along with "instr_table" at 0x7000:
![image](https://user-images.githubusercontent.com/82067599/113794553-cf6d7b00-9718-11eb-863b-49930f9f78fb.png)
Along with "samp_table" at 0x8000:
![image](https://user-images.githubusercontent.com/82067599/113794599-ef9d3a00-9718-11eb-90bd-3e0c60259552.png)
 
Kind of at a lost at this point, whether I need a more specific ROM that I can't seem to find online, or if it's an issue with the program itself, every other Mame ROM I've used (vasv, sfa3, sfa2, msh, mvsc, off the top of my head), has worked fine with it.

If nothing else I have all the files that vgmtrans requires (the .01 and .02, and 11m and 12m), along with the xml file.  I don't know if I can post the whole ROM here but mine is from the WinKawaks site, Euro 950424 version.  
[cybots.zip]()

Anybody know anything or could steer me in the right direction?  Would probably fix it if I could but I'm stumped.

**Comment by apollolux** [comment]
> The seq_table offset seems to be correct at 0x9005

I don't rightly know what types particular MAME offsets expect, but if seq_table is supposed to contain 16-bit values then I would think it starts from 0x9004 or 0x9006 instead, with 24- or 32-bit values starting from 0x9004, right? If the highlighted block is indeed the complete seq_table block then it's 491 bytes, only valid if the seq_table uses 8-bit values (which I have a gut feeling it doesn't) because 492 is divisible by 2, 3, AND 4.

My suggestion is to back up the xml and hand-edit the seq_table offset to 0x9004 instead and try then, reverting to the original if it still doesn't work. Also, FWIW there used to be available a MAME M1-specific set that only had the music data (and I'm pretty sure some corrections) to be played in the M1 Player (I don't know where that is now, unfortunately).

**Reply by EurekaSevven** [reply]
You seem to be correct, upon me looking at it.  And that actually isn't the full size, it starts at 0x9004, and stops at around 0xA005, or 0xA006, after which is the QSound midi/song arrangement data (going off the fact that the data above seems to be a table, hence its name in the xml, and the pattern stops at 0xA006, where it's followed by a 0x21 byte and streams into the song data), so it's more a length of 1003 bytes.  
I only highlighted the areas to show the example of how the offsets line up, really.  
Here's what I mean by the above: 
![image](https://user-images.githubusercontent.com/82067599/113908171-3dab4f80-97a4-11eb-84cf-ba42a5aa0ec3.png)

I'll try what you said in the XML and report back to see if it did anything.  And what you said sounds familiar to me, maybe not the same thing, but it was the QSF/MINIQSF files.

**Reply by EurekaSevven** [reply]
Just tried it, didn't work.

**Comment by mikelow** [comment]
I can't reproduce this. The most-likely issue was a filename or path mismatch of the archived rom files, or the rom archive using something other than standard zip compression.
